AstSemantics.md

```diff
@@ -1,14 +1,22 @@
 # Abstract Syntax Tree Semantics
 
-WebAssembly code is represented as an Abstract Syntax Tree (AST) where each node
-represents an expression. Each function body consists of a list of expressions.
-All expressions and operators are typed, with no implicit conversions or overloading rules.
+WebAssembly code is represented in a form that can be interpreted either as an
+Abstract Syntax Tree (AST) where each node represents an expression, or as a
+typed stack machine, where each node represents an instruction. This document
```

Comment:
Well, you can take a look at the implementation here: https://github.com/WebAssembly/sexpr-wasm-prototype/blob/master/src/wasm-binary-reader-interpreter.c#L1247. It converts from wasm opcodes to a very similar stack machine, but with gotos instead of structured control flow. `IF` becomes `BR_UNLESS`, `ELSE` becomes a `BR` and a branch fixup for `BR_UNLESS`, and `END` does a branch fixup of the `BR`.

The only other slightly tricky part is determining whether the result of the expression needs to be discarded. With the current opcodes, that must be done in a prepass.